sdk/storage/Azure.Storage.Common/src/Shared/Errors.cs

```diff
@@ -46,6 +46,15 @@ public static ArgumentException InvalidService(char s)
         public static ArgumentException InsufficientStorageTransferOptions(long streamLength, long statedMaxBlockSize, long necessaryMinBlockSize)
             => new ArgumentException($"Cannot upload {streamLength} bytes with a maximum transfer size of {statedMaxBlockSize} bytes per block. Please increase the StorageTransferOptions.MaximumTransferSize to at least {necessaryMinBlockSize}.");
 
+        public static InvalidDataException HashMismatch(string hashHeaderName)
+            => new InvalidDataException($"{hashHeaderName} did not match recieved data.");
+
+        public static ArgumentException CannotDeferTransactionalHashVerificationWithClientsideEncryption()
+            => new ArgumentException($"Cannot defer transactional hash verification. Hash goes to pre-decrypted contents, which are unavailable to caller.");
+
+        public static ArgumentException PrecalculatedHashNotSupportedOnSplit()
+            => new ArgumentException("Precalculated hash not supported when potentially partitioning an upload.");
```

Comment:
when do we throw this?

Reply:
Argument validation in PartitionedUploader constructor. Also start of ShareFileClient.UploadInternal, since it doesn't use the partitioned uploader for some reaosn.